lib/src/utils/datetime_utils.dart

```diff
@@ -0,0 +1,19 @@
+class DateTimeUtils {
+  static Duration? parseDuration(dynamic offset) {
+    if (offset == null) return null;
+    if (offset is int) {
+      return Duration(hours: offset);
+    } else if (offset is String) {
+      final regex = RegExp(r'^([+-])(\d{2}):(\d{2})$');
+      final match = regex.firstMatch(offset);
+      if (match == null) return null;
+
+      final sign = match.group(1) == '+' ? 1 : -1;
+      final hours = int.parse(match.group(2)!);
+      final minutes = int.parse(match.group(3)!);
+
+      return Duration(hours: hours * sign, minutes: minutes * sign);
+    }
+    return null;
```

Comment:
IMO, throwing is a dangerous tactic with Health Connect - too much of the undocumented variation in how data is serialized. How about we fallback to logging (as in the update to this PR)?

Reply:
TL;DR: Let's do the what you suggested.

Logging errors at the plugin level is usually not a good idea, because it means people using the plugin cannot do their own error logging.

For example, when I use the plugin in production I want to be able to send myself some error reports if something is not working as expected. And since the errors will happen when a user is using the app I will not see any console logs. The only way for me to track and report errors is if the plugin is returning an error.

But your point is valid, if the data is not as expected returning an error means we don't see any of the data at all, even if the only isssue is with the date, which is far from ideal.

I think the only way to handle this cleanly would be to also return an array of warnings for things that went wrong when the plugin sends a result. But I don't have the time to do that now, so let's just do what you suggested.

Reply:
Thanks!

`dart:developer log` is a convenient stream abstraction over standard console. To collect local logs from our code and from packages that use `dart:developer log`, we, for instance, use `logger` package wired to the log stream. It get noisy sometimes, though.

```
void setupLogListener(Logger logger) {
  developer.logStream.listen((logRecord) {
    // Convert logRecord to logger levels and forward to logger
    switch (logRecord.level.name) {
      case 'SEVERE':
        logger.e(logRecord.message);
        break;
      case 'WARNING':
        logger.w(logRecord.message);
        break;
      case 'INFO':
        logger.i(logRecord.message);
        break;
      default:
        logger.d(logRecord.message);  // Default to debug level
    }
  });
}

Reply:
That's a cool feature I didn't know about, thanks!